File: mobile/library/kotlin/src/io/envoyproxy/envoymobile/grpc/GRPCRequestHeaders.kt
package io.envoyproxy.envoymobile

/*
 * Headers representing an outbound gRPC request.
 */
class GRPCRequestHeaders : RequestHeaders {
  /**
   * Internal constructor used by builders.
   *
   * @param headers: Headers to set.
   */
  internal constructor(headers: Map<String, List<String>>) : super(headers)

  /**
   * Convert the headers back to a builder for mutation.
   *
   * @return GRPCRequestHeadersBuilder, The new builder.
   */
  fun toGRPCRequestHeadersBuilder() = GRPCRequestHeadersBuilder(
    headers.mapValues {
      it.value.toMutableList()
    }.toMutableMap()
  )
}

File: mobile/library/swift/grpc/GRPCRequestHeaders.swift
import Foundation

/// Headers representing an outbound gRPC request.
@objcMembers
public final class GRPCRequestHeaders: RequestHeaders {
  /// Convert the headers back to a builder for mutation.
  ///
  /// - returns: The new builder.
  public func toGRPCRequestHeadersBuilder() -> GRPCRequestHeadersBuilder {
    return GRPCRequestHeadersBuilder(container: self.container)
  }
}
